docs/proposals/timeouts.md

```diff
@@ -0,0 +1,119 @@
+    DeployItem before the deployer has finished its work. Currently, it is completely unclear how long this work
+    might endure.
+
+- Too many possible settings with sometimes strange semantics/implementations
+  - Understanding the timeouts and their exact semantics is very complicated. 
+    - For example Steampunk: They started setting DeployItem.DeployItemSpec.Timeout=15m but where completely surprised 
+      that some of their readiness checks failed because these have a default timeout of 3 minutes. Therefore, they also set 
+      ProviderConfiguration...ReadinessCheckConfiguration.Timeout=15min for every DeployItem. Similar problems might occur
+      for exports.
+    - Even if the overall timeout for a DeployItem is set, no specific timeout is set for the helm chart deployment with 
+      helm 3. What does this mean? How long might the deployer hang in this step and thereby preventing another 
+      reconciliation even after the timeout controller has set the state on failure? Or will the helm 3 deployment fail,
+      due to some other default settings. There are similar questions for the manifest and helm-manifest deployer, e.g.
+      what is the default timeout for the http client etc.?
+  - There are some really strange or even inconsistent implementations e.g. 
+    - Timeout in customreadinesscheck. Checks are executed one after the other in an undefined order. So it depends on 
+      that order how much time was already spend before which might result in a failure of a readiness check timeout.
+    - deleteTimeout is used in the context of manifest and helm manifest deployer when deleting orphaned resources during
+      an upgrade. This setting has nothing to do with the deletion of a DeployItem which is very confusing.
+
+- Pickup timeout: If there is a high load on the system, the pickup of particular DeployItems by a deployer is deferred 
+  because the worker threads might be exhausted and therefore their state is set on failure by the timeout controller. 
+  Just waiting some additional time would have been resulted in their proper processing. This resulted in problems for
+  the Steampunk team.
+
+
+## Solution proposal
+
+- Remove pickup timeout, set it on a very large value or add some DeployItem specific setting:
+  - The main reason for the pickup timeout was to resolve the problem, that if someone wrongly configures a DeployItem
+    such that no deployer would be responsible, that this DeployItem is finished. This use case is mainly relevant during
+    development. Comparing this with the problem of failed DeployItems in productive systems under heavy load, seems to
+    be a bad deal. Furthermore, for the development scenario, it is already possible to find out that a DeployItem
+    is not picked up, by its empty status and such DeployItems could be stopped by the interrupt annotation. 
+    Therefore, the removal of this timeout or setting it on a high value might be a good idea. 
```

Comment:
I disagree here. Besides showing errors during development, the pickup timeout has a second purpose: it detects if the corresponding deployer is not running. Imagine the following scenario: You are using a managed Landscaper service, so you don't have access to the deployer pods. The deployer responsible for your DeployItem was running, so the DI has a status, but then it crashed for whatever reason and is unable to come up again. Now you apply some change to your DI. This change will never get reconciled and to recognize the indicators of the deployer not running you would need detailed knowledge about the Landscaper-internal logic (and even then, most of these indicators could also hint to other failures).
The result: Your changes are not applied and you have no way to find out why.

For this reason, I would strongly advise against removing the pickup timeout. It should already be possible to deactivate it via the Landscaper configuration. 
I don't have a strong opinion about the duration - of course it would be nice to detect configuration errors and missing deployers as fast as possible, but running into pickup timeouts when the system is under load is pretty annoying and should be prevented. Maybe half an hour or an hour would be a good duration - if we hit that under load, it's a strong sign we need to scale up.

Reply:
Is there really such a big difference for a user with no access to the deployer, if the DI remains in a state with a status.transitionTime.initTime not set or that he sees that a pickup timeout has occurred. In both cases the user does only know that no deployer processed the DI so far. Programming against both in e.g. a pipeline, seems to have the same complexity. 

Monitoring that all deployers are running is the responsibility of the health checks which also have the advantage, that they recognise a failing deployer even if no DI is currently processed.